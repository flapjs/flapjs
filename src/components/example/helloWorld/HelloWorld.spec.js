import React from 'react';
import { shallow } from 'enzyme';

import HelloWorld from './HelloWorld.jsx';

let wrapper;

beforeEach(() =>
{
    wrapper = shallow(<HelloWorld/>);
});

describe('<HelloWorld/> rendering', () =>
{
    test('renders correctly', () =>
    {
        expect(wrapper).toMatchSnapshot();
    });
    test('renders one <h1>', () =>
    {
        expect(wrapper.find('h1')).toHaveLength(1);
    });
    test('renders [rainbow] correctly', () =>
    {
        wrapper.setProps({ rainbow: true });
        expect(wrapper).toMatchSnapshot();
    });
});

describe('<HelloWorld/> interaction', () =>
{
    test('should change rainbow when clicked', () =>
    {
        expect(wrapper.state('rainbow')).toBe(false);
        expect(wrapper.find('h1').hasClass('rainbow')).toBe(false);
        wrapper.simulate('click');
        expect(wrapper.state('rainbow')).toBe(true);
        expect(wrapper.find('h1').hasClass('rainbow')).toBe(true);
    });
    test('should toggle rainbow when clicked', () =>
    {
        wrapper.simulate('click');
        expect(wrapper.state('rainbow')).toBe(true);
        expect(wrapper.find('h1').hasClass('rainbow')).toBe(true);
        wrapper.simulate('click');
        expect(wrapper.state('rainbow')).toBe(false);
        expect(wrapper.find('h1').hasClass('rainbow')).toBe(false);
    });
    test('should keep rainbow when set props', () =>
    {
        wrapper.setProps({ rainbow: true });
        expect(wrapper.find('h1').hasClass('rainbow')).toBe(true);
        wrapper.simulate('click');
        expect(wrapper.find('h1').hasClass('rainbow')).toBe(true);
        wrapper.simulate('click');
        expect(wrapper.find('h1').hasClass('rainbow')).toBe(true);
    });
});
