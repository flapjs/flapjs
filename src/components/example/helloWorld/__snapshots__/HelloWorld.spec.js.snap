// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HelloWorld/> rendering renders [rainbow] correctly 1`] = `
<button
  className="helloWorld"
  onClick={[Function]}
>
  <h1
    className="title rainbow "
  >
    Hello 
    World!
  </h1>
</button>
`;

exports[`<HelloWorld/> rendering renders correctly 1`] = `
<button
  className="helloWorld"
  onClick={[Function]}
>
  <h1
    className="title"
  >
    Hello 
    World!
  </h1>
</button>
`;
